app/components/ui-kit/box.js

```diff
@@ -0,0 +1,240 @@
+import Component from '@ember/component';
+import { checkDictionary } from 'travis/utils/ui-kit/assertions';
+import spacingMixin from 'travis/mixins/ui-kit/spacing';
+import { computed } from '@ember/object';
+import { none, not, reads } from '@ember/object/computed';
+
+const DISPLAYS = {
+  BLOCK: 'block',
+  INLINE_BLOCK: 'inline-block',
+  FLEX: 'flex',
+};
+
+const TEXT_ALIGNMENTS = {
+  LEFT: 'left',
+  CENTER: 'center',
+  RIGHT: 'right',
+  JUSTIFY: 'justify',
+};
+
+const LAYERS = {
+  AUTO: 'auto',
+  ZERO: '0',
+  TEN: '10',
+  TWENTY: '20',
+  THIRTY: '30',
+  FORTY: '40',
+  FIFTY: '50',
+};
+
+const SHADOWS = {
+  SM: 'sm',
+  BASE: 'base',
+  NONE: 'none',
+};
+const RADII = {
+  NONE: 'none',
+  SM: 'sm',
+  BASE: 'base',
+  LG: 'lg',
+  FULL: 'full',
+};
+
+// Border
+const BORDER_WIDTHS = {
+  NONE: 'none',
+  ONE: '1',
+};
+const BASE_BORDER_WIDTH = BORDER_WIDTHS.ONE;
+
+// Position
+const POSITION_TYPES = {
+  STATIC: 'static',
+  FIXED: 'fixed',
+  ABSOLUTE: 'absolute',
+  RELATIVE: 'relative',
+  STICKY: 'sticky',
+};
+const POSITION_VALUES = {
+  ZERO: '0',
+  AUTO: 'auto',
+};
+const POSITION_INSETS = {
+  ZERO: '0',
+  X_ZERO: 'x-0',
+  Y_ZERO: 'y-0',
+  AUTO: 'auto',
+  X_AUTO: 'x-auto',
+  Y_AUTO: 'y-auto',
+};
+
+// Component definition
+export default Component.extend(spacingMixin, {
+  tagName: '',
+
+  // Public interface //
+  tag: 'div',
+
+  color: null,
+  display: null,
+  layer: null,
+  radius: null,
+  shadow: null,
+  textAlign: null,
+
+  borderColor: null,
+  borderWidth: null,
+
+  width: null,
+  height: null,
+
+  margin: null,
+  padding: null,
+  position: null,
+
+  // Private //
+  allClasses: computed(
+    function () {
+      return `
+        ${this.colorClass}
+        ${this.displayClass}
+        ${this.layerClass}
+        ${this.radiusClass}
+        ${this.shadowClass}
+        ${this.textAlignClass}
+        ${this.borderColorClass}
+        ${this.borderWidthClass}
+        ${this.widthClass}
+        ${this.heightClass}
+        ${this.marginClasses}
+        ${this.paddingClasses}
+        ${this.positionClasses}
+      `.replace(/\s\s+/g, ' ');
+    }
+  ),
+
+  hasNoColor: none('color'),
+  colorClass: computed('hasNoColor', 'color', function () {
+    return this.hasNoColor ? '' : `bg-${this.color}`;
+  }),
```

Comment:
That sounds like a really good idea! The function needed to also put the relevant prefix in front of the value (in this case the `bg-`), so I ended up calling it `prefix`. I also ended up adding a couple of other optional params to handle various cases, although maybe it would be better to have them split out into separate versions of this function?

Reply:
The current version looks great to me, thanks! 👍